src/lib/plugin_apis/fs.api

```diff
@@ -361,6 +361,62 @@ gchar* bd_fs_get_mountpoint (const gchar *device, GError **error);
  */
 gboolean bd_fs_resize (const gchar *device, guint64 new_size, GError **error);
 
+typedef enum {
+    BD_FS_NO_SUPPORT,
+    BD_FS_OFFLINE,
+    BD_FS_OFFLINEGROW,
+    BD_FS_ONLINE,
+    BD_FS_ONLINEGROW,
+    BD_FS_ONLINE_AND_OFFLINE,
+    BD_FS_ONLINEGROW_AND_OFFLINE,
+    BD_FS_ONLINEGROW_AND_OFFLINEGROW,
+} BDFsOnlineOffline;
+
+/**
+ * bd_fs_can_resize:
+ * @type: the filesystem type to be tested for installed resize support
+ * @mode: (out): how the filesystem allows resizing (e.g. online, offline, growing or not at all)
+ * @required_utility: (out): the utility binary which is required for resizing (if any)
```

Comment:
What is the purpose of returning the utility back? I hope one day we won't need a utility for everything and then this will cause problems.